packages/server/customer-os-api/repository/dashboard_repository.go

```diff
@@ -1516,3 +1517,121 @@ func getCorrectValueType(valueToExtract any) float64 {
 
 	return v
 }
+
+func (r *dashboardRepository) GetDashboardGRRData(ctx context.Context, tenant string, startDate, endDate time.Time) ([]map[string]interface{}, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "DashboardRepository.GetDashboardRetentionRateContractsRenewalsData")
+	defer span.Finish()
+	tracing.SetDefaultNeo4jRepositorySpanTags(ctx, span)
+	span.LogFields(log.Object("startDate", startDate), log.Object("endDate", endDate))
+
+	session := utils.NewNeo4jReadSession(ctx, *r.driver)
+	defer session.Close(ctx)
+
+	dbRecords, err := session.ExecuteRead(ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+
+		queryResult, err := tx.Run(ctx, fmt.Sprintf(
+			`
+					WITH $startDate AS startDate, $endDate AS endDate
+WITH startDate.year AS startYear, startDate.month AS startMonth, endDate.year AS endYear, endDate.month AS endMonth
+WITH RANGE(startYear * 12 + startMonth - 1, endYear * 12 + endMonth - 1) AS monthsRange
+UNWIND monthsRange AS monthsSinceEpoch
+WITH datetime({
+    year: monthsSinceEpoch / 12,
+    month: monthsSinceEpoch %s,
+    day: 1
+}) AS currentDate
+WITH currentDate,
+						 datetime({
+							 year: currentDate.year,
+							 month: currentDate.month,
+							 day: 1,
+							 hour: 0,
+							 minute: 0,
+							 second: 0,
+							 nanosecond: 0o00000000
+						 }) as beginOfMonth,
+     currentDate + duration({months: 1}) - duration({nanoseconds: 1}) as endOfMonth,
+     currentDate + duration({months: 1}) AS startOfNextMonth
+WITH currentDate.year AS year, currentDate.month AS month, beginOfMonth, endOfMonth, startOfNextMonth
+					OPTIONAL MATCH (t:Tenant {name: $tenant})<-[:CONTRACT_BELONGS_TO_TENANT]-(c:Contract_%s)-[:HAS_SERVICE]->(sli:ServiceLineItem_%s)
+WITH year, month, beginOfMonth, endOfMonth, c.id as cid, sli.parentId as sp, sli
+ORDER BY sli.startedAt ASC
+WHERE c.serviceStartedAt IS NOT NULL 
+  AND sli.startedAt < beginOfMonth 
```

Comment:
AND sli.startedAt < beginOfMonth AND (sli.endedAt IS NULL OR sli.endedAt > beginOfMonth)
should be
sli.startedAt < startOfNextMonth AND (sli.endedAt IS NULL OR sli.endedAt >= startOfNextMonth)